```bash
error: builder for '/nix/store/nh85dbk0z0gs066w6zh6g5dznz0v7jn9-chicken-gl-utils-0.8.0.drv' failed with exit code 70;
       last 10 log lines:
       > /build/gl-utils-0.8.0/gl-utils-core.c:13:10: fatal error: epoxy/gl.h: No such file or directory
       >    13 | #include <epoxy/gl.h>
       >       |          ^~~~~~~~~~~~
       > compilation terminated.
       >
       > Error: shell command terminated with non-zero exit status 256: 'gcc' '/build/gl-utils-0.8.0/gl-utils-core.c' -o '/build/gl-utils-0.8.0/gl-utils-core.o' -c  -fno-strict-aliasing -fwrapv -DHAV
E_CHICKEN_CONFIG_H -DC_ENABLE_PTABLES -Os -fomit-frame-pointer -fPIC -DPIC -DC_SHARED -I/build/gl-utils-0.8.0 -I/nix/store/4khq6crr0vmrk73mp92pccnbxfwh4i50-chicken-5.3.0/include/chicken
       >
       > Error: shell command terminated with nonzero exit code
       > 256
       > "sh /build/gl-utils-0.8.0/gl-utils.build.sh"
       For full logs, run 'nix log /nix/store/nh85dbk0z0gs066w6zh6g5dznz0v7jn9-chicken-gl-utils-0.8.0.drv'.
error: 2 dependencies of derivation '/nix/store/3yc1n7sszbnnm6v1i397119wggkdy0x1-chicken-gl-type-0.4.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/aa62qsn953mwxgvbnjm2r9v6kq2xah1r-chicken-glls-0.12.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ah2vv8yjx2y3iy48cq3sn2ysqdif3kd4-chicken-noise-0.2.0.drv' failed to build
error: 6 dependencies of derivation '/nix/store/rnb3nnz275zryj40yhg48clschdssh9p-chicken-hypergiant-0.5.3.drv' failed to build
```

Should we put some libs in `propagatedBuildInputs`?

--- a/pkgs/development/compilers/chicken/5/default.nix
+++ b/pkgs/development/compilers/chicken/5/default.nix
@@ -3,7 +3,7 @@ let
   callPackage = newScope self;
 
   self = with lib; {
-    pkgs = self;
+    inherit callPackage;
 
     fetchegg = { pname, version, sha256, ... }:
       fetchurl {

--- a/pkgs/development/compilers/chicken/5/eggDerivation.nix
+++ b/pkgs/development/compilers/chicken/5/eggDerivation.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, chicken, makeWrapper }:
+{ callPackage, lib, stdenv, chicken, makeWrapper }:
 { name, src
 , buildInputs ? []
 , chickenInstallFlags ? []
@@ -6,15 +6,15 @@
 , ...} @ args:
 
 let
-  overrides = import ./overrides.nix;
+  overrides = callPackage ./overrides.nix { };
   baseName = lib.getName name;
   override = if builtins.hasAttr baseName overrides
    then
      builtins.getAttr baseName overrides
    else
-     {};
+     lib.id;
 in
-stdenv.mkDerivation ({
+(stdenv.mkDerivation ({
   name = "chicken-${name}";
   propagatedBuildInputs = buildInputs;
   nativeBuildInputs = [ makeWrapper ];
@@ -43,4 +43,4 @@ stdenv.mkDerivation ({
   meta = {
     inherit (chicken.meta) platforms;
   } // args.meta or {};
-} // (builtins.removeAttrs args ["name" "buildInputs" "meta"]) // override)
+} // builtins.removeAttrs args ["name" "buildInputs" "meta"]) ).overrideAttrs override

--- a/pkgs/development/compilers/chicken/5/overrides.nix
+++ b/pkgs/development/compilers/chicken/5/overrides.nix
@@ -1,2 +1,58 @@
-{
+{ pkgs, lib, chickenEggs }:
+let
+  addToBuildInputs = pkg: old: {
+    buildInputs = (old.buildInputs or [ ]) ++ lib.toList pkg;
+  };
+  addToPropagatedBuildInputs = pkg: old: {
+    propagatedBuildInputs = (old.propagatedBuildInputs or [ ])
+      ++ lib.toList pkg;
+  };
+  addPkgConfig = old: {
+    nativeBuildInputs = (old.nativeBuildInputs or [ ]) ++ [ pkgs.pkg-config ];
+  };
+  addToBuildInputsWithPkgConfig = pkg: old:
+    (addPkgConfig old) // (addToBuildInputs pkg old);
+  addToPropagatedBuildInputsWithPkgConfig = pkg: old:
+    (addPkgConfig old) // (addToPropagatedBuildInputs pkg old);
+in {
+  breadline = addToBuildInputs pkgs.readline;
+  cairo = old:
+    (addToBuildInputsWithPkgConfig pkgs.cairo old)
+    // (addToPropagatedBuildInputs (with chickenEggs; [ srfi-1 srfi-13 ]) old);
+  cmark = addToBuildInputs pkgs.cmark;
+  dbus = addToBuildInputsWithPkgConfig pkgs.dbus;
+  epoxy = addToPropagatedBuildInputsWithPkgConfig pkgs.libepoxy;
+  exif = addToBuildInputs pkgs.libexif;
+  expat = addToBuildInputs pkgs.expat;
+  ezxdisp = addToBuildInputsWithPkgConfig pkgs.xorg.libX11;
+  freetype = addToBuildInputs pkgs.freetype;
+  glfw3 = addToBuildInputsWithPkgConfig pkgs.glfw3;
+  icu = addToBuildInputsWithPkgConfig pkgs.icu;
+  imlib2 = addToBuildInputsWithPkgConfig pkgs.imlib2;
+  lazy-ffi = addToBuildInputs pkgs.libffi;
+  leveldb = addToBuildInputs pkgs.leveldb;
+  magic = addToBuildInputs pkgs.file;
+  mdh = addToBuildInputs pkgs.pcre;
+  nanomsg = addToBuildInputs pkgs.nanomsg;
+  openssl = addToBuildInputs pkgs.openssl;
+  rocksdb = addToBuildInputs pkgs.rocksdb;
+  sdl-base = addToBuildInputs pkgs.SDL;
+  sdl2 = addToPropagatedBuildInputsWithPkgConfig pkgs.SDL2;
+  sdl2-image = addToBuildInputs pkgs.SDL2_image;
+  sdl2-ttf = addToBuildInputs pkgs.SDL2_ttf;
+  sqlite3 = addToBuildInputs pkgs.sqlite;
+  stemmer = addToBuildInputs pkgs.libstemmer;
+  stfl = addToBuildInputs [ pkgs.ncurses pkgs.stfl ];
+  taglib = addToBuildInputs [ pkgs.zlib pkgs.taglib ];
+  uuid-lib = addToBuildInputs pkgs.libuuid;
+  ws-client = addToBuildInputs pkgs.zlib;
+  xlib = addToPropagatedBuildInputs pkgs.xorg.libX11;
+  yaml = addToBuildInputs pkgs.libyaml;
+  zlib = addToBuildInputs pkgs.zlib;
+  zmq = addToBuildInputs pkgs.zeromq;
+  zstd = addToBuildInputs pkgs.zstd;
+
+  # platform changes
+  pledge = old: { meta = old.meta // { platforms = lib.platforms.openbsd; }; };
+  unveil = old: { meta = old.meta // { platforms = lib.platforms.openbsd; }; };
 }

